osf/utils/fields.py

```diff
@@ -12,7 +12,7 @@ class LowercaseCharField(models.CharField):
     def get_prep_value(self, value):
         value = super(models.CharField, self).get_prep_value(value)
         if value is not None:
-            value = value.lower()
+            value = str(value).lower()
```

Comment:
I'm concerned about letting invalid input pass. How about we revert this change for now so we can diagnose where the invalid input is coming from?